Sources/apm-agent-ios/InstrumentationConfiguration.swift

```diff
@@ -0,0 +1,23 @@
+// Copyright © 2023 Elasticsearch BV
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+import Foundation
+
+
+public struct InstrumentationConfiguration {
+    
+    public var enableAgent = true
+    public var enableCrashReporting = true
```

Comment:
How about adding config options for each of the following to enable / disable?:

- user action instrumentation (i.e. ViewController instrumentation)
- HTTP requests instrumentation
- App metrics
- System metrics (i.e. CPU & Memory)

I think at least for the metrics (App metrics and system metrics) we should make it configurable.